### PR TITLE
[AIRFLOW-2526] params can be overridden through CLI

### PR DESCRIPTION
**Pull request by milton0825**
Make sure you have checked _all_ steps below.

### JIRA
- [x] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-2526
    - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a JIRA issue.

### Description
- [x] Here are some details about my PR, including screenshots of any UI changes:
params can be overridden by the dictionary passed through `airflow backfill -c`

```
templated_command = """
    echo "text = {{ params.text }}"
"""

bash_operator = BashOperator(
    task_id='bash_task',
    bash_command=templated_command,
    dag=dag,
    params= {
        "text" : "normal processing"
    })
```

In daily processing it prints:
```
normal processing
```

In backfill processing `airflow trigger_dag -c "{"text": "override success"}"`, it prints 
```
override success
```

### Tests
- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### Documentation
- [x] In case of new functionality, my PR adds documentation that describes how to use it.
    - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.

### Code Quality
- [x] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Comment by milton0825** [comment]
@Fokko @kaxil @mistercrunch

**Comment by kaxil** [comment]
@milton0825  Can you let me know on where do you use this? Just curious to know why you want to over-ride or the specific use-case for you.

**Comment by milton0825** [comment]
@kaxil we have a use case that we want to override the params during backfill and trigger_dag.

We can pass some key-value pairs through `backfill -c '{"text": "blah blah"}'`
or `trigger_dag -c '{"text": "blah blah"}'`
and it will override the value in {{ params.text }}

**Comment by kaxil** [comment]
LGTM. But as this is a core update it requires approval from 2 committers. Can one of you please take a look as well @Fokko @jgao54 @ashb  ?

File: airflow/models.py [hunk 1]
```diff
@@ -1795,6 +1795,9 @@ def get_template_context(self, session=None):
         if task.params:
             params.update(task.params)
 
+        if dag_run and dag_run.conf:
+            params.update(dag_run.conf)
```
Comment (mistercrunch) [inline airflow/models.py]: I'm not sure about this. `params` is documented and has a clear hierarchy, and nowhere it says that your DAGRun conf will override what is in there. I don't think of the dag_run's configuration as something in the same namespace as `params`. Even if that was the case, should the conf be higher in the override hierarchy? It surely is neither intuitive or documented clearly here.

This is not backward compatible and could introduce rare and intricate collisions/change of behavior.

Plus DagRun.conf is already available in the template through `{{ dag_run.conf }}`, why is there a need to merge it into `params`

Comment (milton0825) [reply airflow/models.py]: @mistercrunch We want to pass some key-value pairs through `backfill -p '{"text": "blah blah"}'`or `trigger_dag -p '{"text": "blah blah"}'` to override the value in `{{ params.text }}`. 
Do you have recommended ways to do that?

**Comment by mistercrunch** [comment]
Then I'd say a new `airflow.cfg` configuration `DAGRUN_CONF_OVERRIDES_PARAMS = false`, where we may switch the default in `2.0`.

**Comment by milton0825** [comment]
@mistercrunch I have added a flag `dag_run_conf_overrides_params` in `airflow.cfg`. can you take a look?

**Comment by mistercrunch** [comment]
LGTM

**Pull request closed without merging**